gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -293,6 +287,8 @@ public synchronized void close() throws IOException {
             closeAPITrace(CLOSE_METHOD, stopwatch);
           }
 
+          streamStats.close();
```

Comment:
Don't we need a similar instrumentation for `close` call as well?

Reply:
close() wont be called this frequently. So, a more readable implementation is better.

Reply:
IMO, It would be better to follow one strategy all across.

Reply:
The other pattern is used at 20+ places. It is useful to have code reuse.